dailystats.py

```diff
@@ -82,12 +83,19 @@ def read_stats_file(csv_file_name):
     result = defaultdict(list)
     with open(csv_file_name, "r") as file:
         rows = list(csv.DictReader(file))
+
+    parse_date = lambda date_str: datetime.strptime(date_str, "%Y-%m-%d")
+    parse_count = lambda count_str: 0 if not count_str else int(count_str)
+

```

Comment:
I noticed we were doing parsing later when computing the results - this seems like it belongs to the part of the code which parses the csv

Reply:
`lambda` is an inline function declaration - for example the following are equivalent:
```python
double = lambda x: x * 2
```
```python
def double(x):
    return x * 2
```